### PR TITLE
Image signing verification

### PR DESCRIPTION
**Pull request by nathanpeck**
# Description

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] [Understand the repository structure](./docs/structure.md)
* [ ] [Add yourself as an author](./docs/add-yourself-as-an-author.md)
* [ ] [Add your pattern](./docs/add-a-pattern.md)
* [ ] [Add a great SVG diagram for your pattern](./docs/svg-diagram-process.md)
* [ ] [Read our general contribution guidelines](./CONTRIBUTING.md)
* [ ] [Read our code of conduct](../CODE_OF_CONDUCT.md)
* [ ] Confirm your pattern frontmatter lints correctly. i.e. run `npm install && npm run lint` locally from your dev environment

### CODE REVIEW

**Review by jldeen — APPROVED** [review]

File: pattern/ecs-fargate-container-image-signing-and-verification/diagram.svg
Comment (iamsamirzon) [inline pattern/ecs-fargate-container-image-signing-and-verification/diagram.svg:1]: Change "Check image manifest" to "Check Image Signature"

File: pattern/ecs-fargate-container-image-signing-and-verification/files/sigverify/sigverify.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+def notation_verify_signature(ecr_client, container_image):
+    auth_token=ecr_client.get_authorization_token()
+    username, password = base64.b64decode(auth_token['authorizationData'][0]['authorizationToken']).decode('utf-8').split(":")
+    # can add --verbose argument to notation verify if needed
+    result=subprocess.run(['notation', 'verify', '-u', username, '-p', password, container_image, '--verbose'])
+    return result
+
+def lambda_handler(event, context):
+
+    # only run when we first provision the container
```
Comment (iamsamirzon) [inline pattern/ecs-fargate-container-image-signing-and-verification/files/sigverify/sigverify.py:24]: should we add a comment that if needed customers can modify this to verify signatures not just at  the initial deploy time, but also at every task launch?

File: pattern/ecs-fargate-container-image-signing-and-verification/index.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+1702484161673,"Error: signature verification failed: no signature is associated with ""209640446841.dkr.ecr.us-east-2.amazonaws.com/bun-hitcounter@sha256:22f34fb040d17d2cca44ba5903c6af24b3cf6ed97bc1aeb257c510b1b829701d"", make sure the artifact was signed successfully
+1702484161674,"209640446841.dkr.ecr.us-east-2.amazonaws.com/bun-hitcounter@sha256:22f34fb040d17d2cca44ba5903c6af24b3cf6ed97bc1aeb257c510b1b829701d failed signature verification.
+1702484161676,"END RequestId: 1883a890-84f1-4ac9-8b4c-ca63c395e0e2
+1702484161676,"REPORT RequestId: 1883a890-84f1-4ac9-8b4c-ca63c395e0e2	Duration: 385.34 ms	Billed Duration: 386 ms	Memory Size: 1024 MB	Max Memory Used: 89 MB
+"
+```
+
+#### Tear it down
+
+When you are done experimenting with this setup you can run the following command to tear down the created infrastructure:
```
Comment (iamsamirzon) [inline pattern/ecs-fargate-container-image-signing-and-verification/index.md:365]:
```suggestion
When you are done experimenting with this setup you can run the following command to tear down the created infrastructure. Indepdently, you may cancel or revoke the signing profile.
```

File: pattern/ecs-fargate-container-image-signing-and-verification/index.md [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+1702417816160,"level=info msg=""Processing signature with manifest mediaType: application/vnd.oci.image.manifest.v1+json and digest: sha256:6228069a242657828cd524c21a554bb7e1f877fad63ff9c9848f3822a470a028"""
+1702417816317,"level=info msg=""Trust policy configuration: &{Name:aws-signer-tp RegistryScopes:[*] SignatureVerification:{VerificationLevel:strict Override:map[]} TrustStores:[signingAuthority:aws-signer-ts] TrustedIdentities:[arn:aws:signer:us-east-2:209640446841:/signing-profiles/SigningProfile_0y9b0jhBJAoh]}"""
+1702417816557,"Successfully verified signature for 209640446841.dkr.ecr.us-east-2.amazonaws.com/aws-signer-image-verification-registrystack-15g6wn192kvr1-signedcontainerregistry-ryrpet5bhnbg@sha256:7ebff78b7d7bd0cb13d462ecf4d9aaa6ea7571bd5548008163d0499eae2fbf40"
+1702417816561,"END RequestId: 25643a4a-147d-406d-bc0a-b2b5c5bae498"
+1702417816561,"REPORT RequestId: 25643a4a-147d-406d-bc0a-b2b5c5bae498	Duration: 910.85 ms	Billed Duration: 1308 ms	Memory Size: 1024 MB	Max Memory Used: 100 MB	Init Duration: 396.69 ms"
+```
+
+For comparison, here is the output for another unsigned image that has no signature:
```
Comment (iamsamirzon) [inline pattern/ecs-fargate-container-image-signing-and-verification/index.md:347]: Could we add a output for another image which is signed but by a different signing profile

File: pattern/ecs-fargate-container-image-signing-and-verification/index.md [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+
+#### Architecture
+
+The following diagram shows the architecture you will deploy:
+
+!!! @/pattern/ecs-fargate-container-image-signing-and-verification/diagram.svg
+
+::: warning
+This EventBridge powered hook is read-only, and asynchronous from the ECS task launch workflow.
+Therefore this hook based architecture is only intended for auditing and notifying of unsigned or unapproved images
```
Comment (iamsamirzon) [inline pattern/ecs-fargate-container-image-signing-and-verification/index.md:35]:
```suggestion
Therefore this hook based architecture is only intended for auditing and notifying of unsigned or unapproved images, great for compliance controls or auditing use cases.
```

**Pull request merged**